loads/src/main/java/org/mousephenotype/cda/loads/cdaloader/steps/MarkerWriter.java

```diff
@@ -0,0 +1,145 @@
+    private final Logger logger = LoggerFactory.getLogger(this.getClass());
+
+    @Autowired
+    @Qualifier("sqlLoaderUtils")
+    private CdaLoaderUtils cdaLoaderUtils;
+
+    private MarkerPSSetter       pss     = new MarkerPSSetter();
+    private Map<String, Integer> written = new HashMap<>();
+
+    public MarkerWriter() {
+
+        written.put("genes", 0);
+        written.put("synonyms", 0);
+        written.put("xrefs", 0);
+    }
+
+    /**
+     * Process the supplied data element. Will not be called with any null items
+     * in normal operation.
+     *
+     * @param items items to be written
+     * @throws Exception if there are errors. The framework will catch the
+     *                   exception and convert or rethrow it as appropriate.
+     */
+    @Override
+    public void write(List items) throws Exception {
+        for (Object genomicFeature1 : items) {
+            GenomicFeature gene = (GenomicFeature) genomicFeature1;
+            pss.setFeature(gene);
+
+            try {
+                Map<String, Integer> counts = cdaLoaderUtils.insertGene(gene, pss);
+                written.put("genes", written.get("genes") + counts.get("genes"));
+                written.put("synonyms", written.get("synonyms") + counts.get("synonyms"));
+                written.put("xrefs", written.get("xrefs") + counts.get("xrefs"));
+
+            } catch (Exception e) {
+                throw new CdaLoaderException(e.getLocalizedMessage() + "\n\t gene: " + gene);
+            }
+        }
+    }
+
+    public class MarkerPSSetter implements PreparedStatementSetter {
+        private GenomicFeature feature;
+
+        public void setFeature(GenomicFeature feature) {
+            this.feature = feature;
+        }
+
+        @Override
+        public void setValues(PreparedStatement ps) throws SQLException {
+            Integer biotypeDbId = (feature.getBiotype() == null ? null : feature.getBiotype().getId().getDatabaseId());
+            String biotypeAcc = (feature.getBiotype() == null ? null : feature.getBiotype().getId().getAccession());
+            Integer subtypeDbId = (feature.getSubtype() == null ? null : feature.getSubtype().getId().getDatabaseId());
+            String subtypeAcc = (feature.getSubtype() == null ? null : feature.getSubtype().getId().getAccession());
+
+//          INSERT INTO genomic_feature (acc, db_id, symbol, name, biotype_acc, biotype_db_id, subtype_acc, subtype_db_id, seq_region_id, seq_region_start, seq_region_end, seq_region_strand, cm_position, status)
+//          VALUES (?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?, ?)
+
+            ps.setString(1, feature.getId().getAccession());                // acc
+            ps.setInt(2, feature.getId().getDatabaseId());                  // db_id
+            ps.setString(3, feature.getSymbol());                           // symbol
+            ps.setString(4, feature.getName());                             // name
+            if (biotypeAcc == null) {
```

Comment:
Take a look into SqlUtils class.  It has convenience methods for handling null values